components/form/CheckboxGroup.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { useFormContext } from 'react-hook-form';
+import PropTypes from 'prop-types';
+import { CheckboxLabel, ErrorLabel } from '.';
+
+export function Checkbox({ ...props }) {
+  const { register } = useFormContext();
+  return (
+    <>
+      <div className={`flex items-center ${props.className}`}>
+        <input
+          id={`checkbox-${props.value}`}
+          className="w-4 h-4 text-amcovad-primary-400 border-gray-200 rounded focus:ring-amcovad-primary-300 "
+          name={props.name}
+          type="checkbox"
+          {...register(props.name)}
+          value={props.value}
+        />
+        <CheckboxLabel htmlFor={`checkbox-${props.value}`} label={props.label} />
```

Comment:
This should just be `Label`. It doesn't make sense to create a different Label for Checkbox.

Also, instead of setting `htmlFor` here, we should pass in a `name` props and use the value of that name as the `htmlFor`.

```suggestion
        <Label name={`checkbox-${props.value}`} label={label} />
```